.github/workflows/status.yaml

```diff
@@ -0,0 +1,64 @@
+name: Custom GH Status from Workflow Artifacts
+
+on:
+  workflow_run:
+    workflows: ["pr"]
+    types:
+      - completed
+
+jobs:
+  process_artifacts:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: gh-status
+          path: downloaded_artifacts/
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Read artifact JSON
+        id: read_artifact
+        run: |
+          CONTENT=$(cat downloaded_artifacts/gh-status.json)
+          echo "content=$CONTENT" >> $GITHUB_OUTPUT
+      - name: Create status
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ARTIFACT_CONTENT: ${{ steps.read_artifact.outputs.content }}
+          COMMIT_SHA: ${{ github.event.workflow_run.head_sha }}
+          WORKFLOW_RUN_ID: ${{ github.event.workflow_run.id }}
+          REPO_NAME: ${{ github.repository }}
+          ATTEMPTS: ${{ github.event.workflow_run.run_attempt }}
+        run: |
+          JOB_ID=$(gh api repos/$REPO_NAME/actions/runs/$WORKFLOW_RUN_ID/jobs --paginate --jq '.jobs[] | select(.name == "pandas-tests-diff / build") | .id')
+          CONTENT_JSON=$(echo $ARTIFACT_CONTENT | jq .)
+          CUSTOM_CONTEXT=$(echo $CONTENT_JSON | jq -r .context)
+          CUSTOM_DESCRIPTION=$(echo $CONTENT_JSON | jq -r .description)
+          CUSTOM_TARGET_URL=$(echo $CONTENT_JSON | jq -r .target_url)
+          CUSTOM_STATE=$(echo $CONTENT_JSON | jq -r .state)
```

Comment:
If we can shellcheck on this script, it would complain that all of these `$CONTENT_JSON` variables aren't quoted.

We should add some quotes here and check the rest of the file as well.

Reply:
Yup, done 👍